File: Code Fights/Almost Increasing Sequence/AlmostIncreasingSequence.cs
bool almostIncreasingSequence(int[] sequence)
{
 
    if (sequence.Length == 2)
        return true;
    
    int counter = 0;
    int counterTwo = 0;
    for (int i = 0; i < sequence.Length - 1; i++)
    {
        if (sequence[i] >= sequence[i+1])
           counter++;
        
        
        if(i != 0) 
        {
           if (sequence[i - 1] >= sequence[i + 1]) 
               counterTwo++;
        }
    }
    
    if (counter == 1 && counterTwo <= 1)
        return true;
    else 
        return false;
}
    

File: Code Fights/Almost Increasing Sequence/AlmostIncreasingSequence.c
bool almostIncreasingSequence(arr_integer sequence)
{
    int n = 0, i = 0;
    
    while(i < sequence.size-1) {
        if(sequence.arr[i] < sequence.arr[i+1])
            i++;
        else
        {
            n++;
            if( i > 0 && i+2 < sequence.size && sequence.arr[i+1] <= sequence.arr[i-1] && sequence.arr[i+2] <= sequence.arr[i]) 
                n++;
            else 
                i++;
            
            if (n > 1)
                return false;
        }
    }
    return true;
}

